libpod/container_api.go

```diff
@@ -776,3 +776,21 @@ func (c *Container) ShouldRestart(ctx context.Context) bool {
 	}
 	return c.shouldRestart()
 }
+
+// ResolvePath resolves the specified path on the container.  The path may
+// resolve to the container's mount point or to a volume or bind mount.  It
+// returns the root (e.g., container, volume or bind mount) and the fully
+// resolved path which is either equal to or a subdirectory of the root.
+func (c *Container) ResolvePath(ctx context.Context, path string) (string, string, error) {
```

Comment:
This should do the standard lock + sync if it's a public function

Reply:
It's safe to use concurrently/

Reply:
Or shall I lock and then `mount` and `unmount`?

Reply:
My concern isn't really concurrency so much as removed containers - we could have the container removed out from under us midway through this, between Mount and Unmount, for example. Preference would be lock then `mount()` / `unmount()`

Reply:
Although, looking at it more, I think a lot of the logic in `Unmount()` isn't duplicated in `unmount()`... I don't think it can be safely moved in there, either (we have two cases for unmounting - cases like this, where an external user wants to access the container's root filesystem - the code in Unmount() works there - and cases where Libpod wants to stop the container - the code in Unmount works there).

@vrothberg What would you think about making `ResolvePath()` be contingent on the container already being mounted? It seems like it would simplify the code, and potentially make things a bit more performant (we wouldn't need to call Mount more than once in the Copy code).

Reply:
@mheon sounds good to me.  One early version of the change was exactly like that but I preferred to play it safe.  I'll go as suggested and repush.